CorsixTH/Lua/world.lua

```diff
@@ -2502,6 +2494,40 @@ function World:afterLoad(old, new)
     -- Move existing and future earthquakes to the Earthquake class
     self.earthquake = Earthquake(self, true)
   end
+  if old < 208 then -- Adjust game speed and tick rates
+    local old_tick_rates
+    if old < 183 then
+      old_tick_rates = {
+        ["Pause"]              = {0, 1},
+        ["Slowest"]            = {1, 9},
+        ["Slower"]             = {1, 5},
+        ["Normal"]             = {1, 3},
+        ["Max speed"]          = {1, 1},
+        ["And then some more"] = {3, 1},
+      }
+    else
+      old_tick_rates = {
+        ["Pause"]              = {0, 1},
+        ["Slowest"]            = {1, 28},
+        ["Slower"]             = {1, 6},
+        ["Normal"]             = {2, 3},
+        ["Max speed"]          = {1, 1},
+        ["And then some more"] = {2, 1},
+        ["Speed Up"]           = {4, 1},
+      }
+    end
+    local found_speed = false
+    for name, rate in pairs(old_tick_rates) do
+      if rate[1] == self.hours_per_tick and rate[2] == self.tick_rate then
+        found_speed = true
+        self:setSpeed(name)
+        break
+      end
+      if not found_speed then
+        self:setSpeed("Normal")
+      end
```

Comment:
I assume this block is meant to be outside the for loop? It's currently inside which means that it executes the first iteration on the loop.